app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java

```diff
@@ -133,6 +141,45 @@ public static DiningHallModel fromJSON(Context context, boolean hardcoded, JSONO
 		return items;
 	}
 
+	@Override
+	public void parseEatery(Context context, boolean hardcoded, AllEateriesQuery.Eatery eatery){
+		super.parseEatery(context, hardcoded, eatery);
+		mWeeklyMenu = new HashMap<>();
+		mSortedDates = new ArrayList<>();
+		DateTimeFormatter timeFormatter = new DateTimeFormatterBuilder()
+				.parseCaseInsensitive()
+				.appendPattern("h:mma")
+				.toFormatter();
+		DateTimeFormatter dateFormatter = new DateTimeFormatterBuilder()
+				.parseCaseInsensitive()
+				.appendPattern("yyyy-MM-dd")
+				.toFormatter();
+
+		mId = eatery.id();
+
+		// Each Operating Hour is a single day for dining halls
+		for(AllEateriesQuery.OperatingHour operatingHour : eatery.operatingHours()){
+			DailyMenuModel dailyMenuModel = new DailyMenuModel();
+			LocalDate localDate = LocalDate.parse(operatingHour.date().toUpperCase());
+			for(AllEateriesQuery.Event event : operatingHour.events()){
+				LocalDateTime start = null, end = null;
+				start = LocalTime.parse(event.startTime().toUpperCase().substring(11),
+						timeFormatter).atDate(localDate);
+				end = LocalTime.parse(event.endTime().toUpperCase().substring(11),
+						timeFormatter).atDate(localDate);
+				MealType type = MealType.fromDescription(event.description());
+				MealModel mealModel = new MealModel(start,end);
+				mealModel.setType(type);
+				MealMenuModel mealMenuModel = MealMenuModel.fromGraphQL(event.menu());
+				mealModel.setMenu(mealMenuModel);
+				if (type != null) {
+					dailyMenuModel.addMeal(type, mealModel);
```

Comment:
i've noticed that indentation is always 8, @ningning621 is this intentional?

Reply:
I'll have to look at the codebase. I think we ran a linter last semester and it messed up some of the spacing. I'm pretty certain that the 8 space indentation shows up only on github and not in the IDE.